### PR TITLE
PR Policy

### PR DESCRIPTION
**Pull request by TheGlitch76**
Uses RFC 39 to define how PRs are accepted into QSL.

Some notes of how this affects the QSL team:
### Powers of the QSL team

- Approve urgent PRs (exclusive)
- Add and remove members of subteams (exclusive)
- Request changes in PRs to block them from being merged until the issues are addressed
- Merge accepted PRs that have left the final comment period (exclusive)
- Join as many or as few subteams as they'd like

The QSL team cannot just merge whatever they want; they must go through this process (with the exception of trivial fixes)

We will have a bot eventually that will manage the final comment period stuff

### CODE REVIEW

**Review by Leo40Git — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,6 +1,38 @@
 # Contributing
 
 This document outlines best-practices and contributing guidelines to the Quilt Standard Libraries.
+## Pull Request Process
+To get a pull request merged into QSL, it must get a certain number of approvals from the maintainers of each library the pull request targets, and then it will enter a Final Comment Period. If the pull request passes the final comment period without opposition, the PR will be merged. Otherwise, the PR will return to being in review.
```
Comment (sylv256) [inline CONTRIBUTING.md]: Who are the "maintainers" of each library?

Comment (sylv256) [reply CONTRIBUTING.md]: Oh, did you make the RFC yet? :confused:

Comment (sylv256) [reply CONTRIBUTING.md]: Nevermind. The link you provided 404s because it doesn't link to a PR.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,6 +1,38 @@
...
+For pull requests that make internal refactors and do not change any API, such as bugfixes.
+
+**Required Approvals**: 1
+- At least 1 approval must come directly from each library team whos code the pull request modifies.
+
+**Final Comment Period**: 3 days
+
+### `T: urgent`
+#### Description
+For pull requests that must be merged quickly, like ports of critical core modules or game-breaking bug fixes.
```
Comment (i509VCB) [inline CONTRIBUTING.md]: Probably a good idea to add anything related to publishing and distributing of qsl in buildscripts.

**Review by i509VCB — APPROVED** [review]

**Review by OroArmor — APPROVED** [review]

**Review by LemmaEOF — APPROVED** [review]
LGTM/LGTM

**Comment by TheGlitch76** [comment]
Hello,
I've updated the PR policy to reflect that the current QSL team does not have all of its subteam positions full. With the new version, an approval from a member of the QSL Core team overrides the requirement for an approval by a specific subteam. I recommend reconsidering this after QSL's release, when the QSL team is (hopefully) much larger.

**Review by LambdAurora — CHANGES_REQUESTED** [review]
Please target the 1.18 branch.

> With the new version, an approval from a member of the QSL Core team overrides the requirement for an approval by a specific subteam. I recommend reconsidering this after QSL's release, when the QSL team is (hopefully) much larger.

I would formulate this in a much more general manner, what about formulating it in a way that doesn't require further modification, like: if the affected team has one member or less, then a core team member can provide the missing requirements.

**Comment by TheGlitch76** [comment]
> if the affected team has one member or less, then a core team member can provide the missing requirements.

I like this idea, only one issue: what do we do in cases where a team officially has two or more members, but they are no longer active contributors? This isn't something I considered when I originally wrote this policy; it just struck me.

**Review by OroArmor — APPROVED** [review]

**Comment by LambdAurora** [comment]
> I like this idea, only one issue: what do we do in cases where a team officially has two or more members, but they are no longer active contributors? This isn't something I considered when I originally wrote this policy; it just struck me.

If after 7 days of non-response, or if the contributor gives permission, QSL core members could override?

**Comment by TheGlitch76** [comment]
I added a clause that the QSL core team may waive the requirement for a subteam review if the team is inactive. QSL Core isn't a zero-trust team, so I think it will work fine.
This whole process is just a guideline and basically unenforceable in a strict sense anyway

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -166,6 +225,7 @@ Use `{@index}` to allow enclosed text to be indexed by the Javadoc search.
 Javadoc will take the first sentence, ended by the first `.`, as a brief description of the member you are documenting.
 Note that `.` from abbreviations, such as `i.e.`, count.
 
+
```
Comment (TheGlitch76) [inline CONTRIBUTING.md]:
```suggestion
```

**Comment by TheGlitch76** [comment]
I added a proper guide to the PR policy, explaining the steps to getting a PR merged and how triage categories etc. work

**Review by LambdAurora — APPROVED** [review]

**Comment by TheGlitch76** [comment]
The FCP for this policy will end in 7 days, assuming no major concerns are raised.

**Review by SilverAndro — APPROVED** [review]

**Pull request merged**